tsl/test/expected/chunk_column_stats.out

```diff
@@ -0,0 +1,282 @@
+-- due to chunk exclusion using min/max ranges calculated above
+:PREFIX SELECT * FROM sample_table WHERE sensor_id > 9;
+          QUERY PLAN          
+------------------------------
+ Seq Scan on _hyper_1_2_chunk
+   Filter: (sensor_id > 9)
+(2 rows)
+
+-- Newly added chunks should also have MIN/MAX entry
+\set start_date '2024-01-28 01:09:51.583252+05:30'
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 1, 9.6054, 78.999, 'new row4');
+SELECT * from _timescaledb_catalog.chunk_column_stats;
+ id | hypertable_id | chunk_id | column_name | column_type |     range_start      |      range_end      | valid 
+----+---------------+----------+-------------+-------------+----------------------+---------------------+-------
+  1 |             1 |        0 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
+  3 |             1 |        2 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
+  2 |             1 |        1 | sensor_id   | integer     |                    1 |                   9 | t
+  4 |             1 |        4 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
+(4 rows)
+
+-- use the disable_column_stats API to remove the min/max range entries
+SELECT * from disable_column_stats('sample_table', 'sensor_id');
+ hypertable_id | column_name | dropped 
+---------------+-------------+---------
+             1 | sensor_id   | t
+(1 row)
+
+SELECT * from _timescaledb_catalog.chunk_column_stats;
+ id | hypertable_id | chunk_id | column_name | column_type | range_start | range_end | valid 
+----+---------------+----------+-------------+-------------+-------------+-----------+-------
+(0 rows)
+
+SELECT * from disable_column_stats('sample_table', 'sensor_id', true);
+NOTICE:  dimension range entry does not exist for column "sensor_id", skipping
+ hypertable_id | column_name | dropped 
+---------------+-------------+---------
+             1 | sensor_id   | f
+(1 row)
+
+\set ON_ERROR_STOP 0
+SELECT * from disable_column_stats('sample_table', 'sensor_id');
+ERROR:  dimension range entry does not exist for column "sensor_id"
+-- should only work on columns that have been enabled for tracking
+SELECT * from disable_column_stats('sample_table', 'time');
+ERROR:  dimension range entry does not exist for column "time"
+SELECT * from disable_column_stats('sample_table', 'cpu');
+ERROR:  dimension range entry does not exist for column "cpu"
+\set ON_ERROR_STOP 1
+-- Check that a DROP COLUMN removes entries from catalogs as well
+SELECT * FROM enable_column_stats('sample_table', 'sensor_id');
+ column_stats_id | created 
+-----------------+---------
+               5 | t
+(1 row)
+
+SELECT decompress_chunk(:'CH_NAME');
+            decompress_chunk            
+----------------------------------------
+ _timescaledb_internal._hyper_1_1_chunk
+(1 row)
+
+-- Entry should be removed for this chunk now
```

Comment:
@erimatnor 
That's a wrong comment. The entry is indeed updated to -inf/+inf. 

Once a chunk is decompressed DML can freely happen into it. We don't have the infrastructure/logic to track the first DML entry point at which we could invalidate today. So to err on the side of caution, we invalidate immediately on decompression for now. Maybe future versions could implement this delayed invalidation.

Reply:
My point was that the entry _doesn't_ have to be set it to -inf/+inf because the constraint is still valid after decompression. So, the idea I alluded to is to change it so that it gets invalidated when the actual data changes.

I am not sure we _don't_ have the infra for it (we do DML decompression on the same kind of insert/updates/deletes, for example). But it is OK to not have this for the first version if you decide it is not worth the effort. 

Just want to point out, again ,though,  the confusion around "invalidation" vs setting infinite range vs deleting. In the test, the range seems to be set to -inf/+inf, but the entry is still marked as "valid". You also mention yourself, in the comment, that the entry is invalidated,  but it is still has valid=true. IMO, this is confusing.

I am actually not sure what the point is in keeping the entry vs just deleting it. Are invalidated or infinite range entries actually used for anything? What is the advantage of have the invalid entry in the table vs just adding a new one once the chunk is compressed again?

Reply:
The entries are used for chunk exclusion. If we don't have these entries then we will have to add new logic in the code to take a list of all chunks, remove chunks that have entries in this catalog and include chunks that don't. The entire thing doesn't sound atomic or consistent to me in terms of chunk selection. With these entries present, we have a consistent view of the chunks range metadata and can use it uniformly IMO.